browser/ui/toolbar/brave_app_menu_model.cc

```diff
@@ -185,6 +186,12 @@ void BraveAppMenuModel::Build() {
   BuildBrowserSection();
   BuildMoreToolsSubMenu();
   BuildHelpSubMenu();
+
+  ApplyLeoIcons(this);
+  ApplyLeoIcons(bookmark_sub_menu_model());
+  for (const auto& submenu : sub_menus()) {
+    ApplyLeoIcons(submenu.get());
+  }
```

Comment:
How about making `ApplyLeoIcons()` as recursive method instead of handling each submenus separately?

Reply:
Unfortunately, `GetSubmenuModelAt` returns a `ui::MenuModel` which doesn't have a `SetIcon` function. I could do a `static_cast` but I don't know if the menus will ever contain something which isn't a `ui::SimpleMenu`.

In addition, we only really apply icons to the first level of submenus, and some menus have arbitrary depth (like the bookmarks menu).

Reply:
ok got it.